cirq/contrib/qcircuit/qcircuit_diagram.py

```diff
@@ -47,15 +48,39 @@ def _render(diagram: circuits.TextDiagramDrawer) -> str:
 
     diagram2 = circuits.TextDiagramDrawer()
     for y in range(h):
+        # have a none wire item on this row
+        row_has_item = False
+        # current horizontal line has something that need to be connected
         for x in range(max(0, w - 1)):
             key = (x, y)
             diagram_text = diagram.entries.get(key)
-            v = '&' + (diagram_text.text if diagram_text else  '') + ' '
+            v = '&' + (diagram_text.text if diagram_text else '') + ' '
+            if diagram_text:
+                row_has_item = True
             diagram2.write(2*x + 1, y, v)
-            post1 = r'\qw' if key in qw else ''
-            post2 = r'\qwx' if key in qwx else ''
-            diagram2.write(2*x + 2, y, post1 + post2)
-        diagram2.write(2*w - 1, y, r'&\qw\\')
+            # if has none wire before, should draw wire
+            post1 = r'\qw' if key in qw and row_has_item else ''
+            post2 = ''
+            if key in qwx:
```

Comment:
This solution is interesting, but it is very hard to read and reason about. 
The key issue is that the moment groups character is not handled well by qcircuit. So, I would rather introduce a by default True parameter to `to_text_diagram_drawer`, something like `draw_moment_groups` - and then the qcircuit drawer could set that to false.

Reply:
Got it! I will get back to this ASAP but I just lost power here a couole hours ago. The storm hit us hard. Just a heads up that I might not be able to work at all in the next day or two

Reply:
No worries, no rush - good luck with the storm, stay safe!

Reply:
Hello @balopat, as I looked at your suggestion, I realized how silly my method is. I tried to keep the angle bracket to indicate the moment as I thought it is a really good reminder for viewers that these gates can be run simultaneously. But I hit a wall and got no good results.  So, I basically did what you said, and added a test in qcircuit_test.py to account for the circuit that caused our initial problem #3183.